lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -2128,8 +2128,69 @@ TypeSystemSwiftTypeRef::GetByteStride(opaque_compiler_type_t type,
 
 lldb::Encoding TypeSystemSwiftTypeRef::GetEncoding(opaque_compiler_type_t type,
                                                    uint64_t &count) {
-  return m_swift_ast_context->GetEncoding(ReconstructType(type), count);
+  if (!type)
+    return lldb::eEncodingInvalid;
+
+  auto impl = [&]() -> lldb::Encoding {
+    count = 1;
+
+    using namespace swift::Demangle;
+    Demangler dem;
+    auto *node = DemangleCanonicalType(dem, type);
+    auto kind = node->getKind();
+
+    if (kind == Node::Kind::BuiltinTypeName) {
+      assert(node->hasText());
+      if (node->getText().startswith(swift::BUILTIN_TYPE_NAME_INT) ||
+          node->getText() == swift::BUILTIN_TYPE_NAME_WORD)
+        return lldb::eEncodingSint;
+      if (node->getText() == swift::BUILTIN_TYPE_NAME_FLOAT)
```

Comment:
We should probably also throw in `swift::BUILTIN_TYPE_NAME_FLOAT_PPC` for good measure.